lib/Transport.js

```diff
@@ -261,6 +265,13 @@ class Transport {
           return onConnectionError(
             new RequestAbortedError(`The content length (${contentLength}) is bigger than the maximum allowed string (${MAX_STRING_LENGTH})`, result)
           )
+        } else if (this.memoryCircuitBreakerEnabled && (contentLength + heapUsed > HEAP_SIZE_LIMIT * this.responseMaxHeapPercentage)) {
+          // Memory circuit breaker enabled and the response payload is too large to fit into available heap memory.
+          // Abort this response to avoid OOM crash of dashboards.
+          response.destroy()
+          return onConnectionError(
+            new RequestAbortedError(`The content length (${contentLength}) is bigger than the maximum allowed heap memory limit`, result)
```

Comment:
Neat: would be great to add the settings they can tune this setting. i.e `You could tune this via `config.name`

Reply:
Yes, I will have a follow up PR to add setting in OSD config (opensearch_dashbaords.yml). This PR is only for change in client repo.

Reply:
Will upgrade the Error message as well to add more instruction guidance